### PR TITLE
rename classes to enforce BugPatternNaming

### PR DESCRIPTION
**Pull request by XN137**
- keep old class names als altNames for backwards compatibility
- prefer buildDescription to keep things DRY

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -74,8 +74,7 @@ tasks {
         // enforce errorprone-specific checks
         options.errorprone.check("MemoizeConstantVisitorStateLookups", CheckSeverity.ERROR)
         options.errorprone.check("ASTHelpersSuggestions", CheckSeverity.ERROR)
-        // our bugpattern classes use custom names
-        options.errorprone.check("BugPatternNaming", CheckSeverity.OFF)
+        options.errorprone.check("BugPatternNaming", CheckSeverity.ERROR)
```
Comment (XN137) [inline build.gradle.kts:77]: i think i've adjusted all the files/spots for the renames:
```
~/code/errorprone-slf4j$ find . | rg "(ManuallyProvidedMessage|FormatShouldBeConst|IllegalPassedClass|PreferFinalSlf4jLogger|DoNotUseStaticSlf4jLogger|DoNotPublishSlf4jLogger|PlaceholderMismatch|SignOnlyFormat)"
./src/test/java/jp/skypencil/errorprone/slf4j/Slf4jSignOnlyFormatTest.java
./src/test/java/jp/skypencil/errorprone/slf4j/Slf4jPlaceholderMismatchTest.java
./src/test/java/jp/skypencil/errorprone/slf4j/Slf4jFormatShouldBeConstTest.java
./src/test/java/jp/skypencil/errorprone/slf4j/Slf4jIllegalPassedClassTest.java
./src/main/java/jp/skypencil/errorprone/slf4j/Slf4jSignOnlyFormat.java
./src/main/java/jp/skypencil/errorprone/slf4j/Slf4jIllegalPassedClass.java
./src/main/java/jp/skypencil/errorprone/slf4j/Slf4jPlaceholderMismatch.java
./src/main/java/jp/skypencil/errorprone/slf4j/Slf4jFormatShouldBeConst.java

~/code/errorprone-slf4j$ rg -w "(ManuallyProvidedMessage|FormatShouldBeConst|IllegalPassedClass|PreferFinalSlf4jLogger|DoNotUseStaticSlf4jLogger|DoNotPublishSlf4jLogger|PlaceholderMismatch|SignOnlyFormat)"
src/main/java/jp/skypencil/errorprone/slf4j/Slf4jLoggerShouldBePrivate.java
23:    altNames = {"DoNotPublishSlf4jLogger"},

src/main/java/jp/skypencil/errorprone/slf4j/Slf4jDoNotLogMessageOfExceptionExplicitly.java
19:    altNames = {"ManuallyProvidedMessage"},

src/main/java/jp/skypencil/errorprone/slf4j/Slf4jFormatShouldBeConst.java
18:    altNames = {"FormatShouldBeConst"},

src/main/java/jp/skypencil/errorprone/slf4j/Slf4jLoggerShouldBeNonStatic.java
24:    altNames = {"DoNotUseStaticSlf4jLogger"},

src/main/java/jp/skypencil/errorprone/slf4j/Slf4jPlaceholderMismatch.java
21:    altNames = {"PlaceholderMismatch"},

src/main/java/jp/skypencil/errorprone/slf4j/Slf4jLoggerShouldBeFinal.java
22:    altNames = {"PreferFinalSlf4jLogger"},

src/main/java/jp/skypencil/errorprone/slf4j/Slf4jSignOnlyFormat.java
18:    altNames = {"SignOnlyFormat"},

src/main/java/jp/skypencil/errorprone/slf4j/Slf4jIllegalPassedClass.java
33:    altNames = {"IllegalPassedClass"},
```

File: src/main/java/jp/skypencil/errorprone/slf4j/Slf4jDoNotLogMessageOfExceptionExplicitly.java [hunk 2]
```diff
@@ -49,11 +50,8 @@ public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState
             .filter(meth -> THROWABLE.test(meth.sym.owner.toString()))
             .findFirst();
     if (problem.isPresent()) {
-      return Description.builder(
-              tree,
-              "Slf4jDoNotLogMessageOfExceptionExplicitly",
-              "https://github.com/KengoTODA/findbugs-slf4j#slf4j_manually_provided_message",
-              ERROR,
+      return buildDescription(tree)
```
Comment (XN137) [inline src/main/java/jp/skypencil/errorprone/slf4j/Slf4jDoNotLogMessageOfExceptionExplicitly.java:53]: `BugPattern.buildDescription` builds the description with defaults from the `@BugPattern` annotation

File: src/main/java/jp/skypencil/errorprone/slf4j/Slf4jLoggerShouldBeNonStatic.java [hunk 3]
```diff
@@ -38,14 +38,7 @@ public Description matchVariable(VariableTree tree, VisitorState state) {
       SuggestedFixes.removeModifiers(tree, state, Modifier.STATIC).ifPresent(builder::merge);
       suggestRename(tree, state).ifPresent(builder::merge);
 
-      return Description.builder(
-              tree,
-              "Slf4jLoggerShouldBeNonStatic",
-              "https://github.com/KengoTODA/findbugs-slf4j#slf4j_logger_should_be_non_static",
-              SUGGESTION,
-              "Do not use static Logger field, use non-static one instead")
```
Comment (XN137) [inline src/main/java/jp/skypencil/errorprone/slf4j/Slf4jLoggerShouldBeNonStatic.java:46]: message will be taken from BugPattern summary if not explicitly set on the description

**Comment by KengoTODA** [comment]
@XN137 LGTM, thanks for your contribution!

I think we do not need to make a minor/patch release for this fix, because we just renamed classes and there is no API change nor new feature. Is my understanding correct?

**Review by KengoTODA — APPROVED** [review]

**Comment by XN137** [comment]
thanks for the review, happy new year ;)

> I think we do not need to make a minor/patch release for this fix, because we just renamed classes and there is no API change nor new feature. Is my understanding correct?

yeah this PR should not change APIs or Behavior except for having the new default and alternative bug checker names.
if we were to completely drop the old names (i.e. not keep `altNames` property) this would be a backward incompatible change.

that being said i would have thought https://github.com/KengoTODA/errorprone-slf4j/commit/4cd613d6ba1b9efd57a3829e761a8e28f78d6b9b might have warranted a more prominent version bump since it changed the required java version and also made a few checks stricter - but its probably no big deal either way.

**Comment by KengoTODA** [comment]
Thanks for your response! I understood, and yes Java 9 & 10 are already EOL then it won't be big deal. Then I'll merge this PR with the `chore:` commit comment header, that triggers no version change.

**Pull request merged**